django_restql/mixins.py

```diff
@@ -552,4 +553,147 @@ def update(self, instance, validated_data):
             fields["many_to"]["one_related"]
         )
 
-        return super().update(instance, validated_data)
\ No newline at end of file
+        return super().update(instance, validated_data)
+
+
+class RestQLViewMixin(object):
+    @property
+    def restql_query(self):
+        serializer_class = self.get_serializer_class()
+
+        if hasattr(serializer_class, "query_param_name"):
+            return serializer_class.get_parsed_query_from_req(self.request)
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        if self.restql_query is not None:
+            queryset = self.get_restql_queryset(queryset)
+        return queryset
+
+    def get_prefetch_related_mapping(self):
+        serializer_class = self.get_serializer_class()
+        if hasattr(self, "prefetch_related"):
+            return self.prefetch_related
+        elif (
+            serializer_class
+            and hasattr(serializer_class, "Meta")
+            and hasattr(serializer_class.Meta, "prefetch_related")
+        ):
+            return self.get_serializer_class().Meta.prefetch_related
+
+        return {}
+
+    def get_select_related_mapping(self):
+        serializer_class = self.get_serializer_class()
+        if hasattr(self, "select_related"):
+            return self.select_related
+        elif (
+            serializer_class
+            and hasattr(serializer_class, "Meta")
+            and hasattr(serializer_class.Meta, "select_related")
+        ):
+            return self.get_serializer_class().Meta.select_related
+
+        return {}
+
+    def get_restql_query_dict(self, data=None):
+        """
+        Returns the RestQL query as a dict.
+        """
+        keys = {}
+        if not data:
+            data = self.restql_query
+
+        for item in data:
+            if isinstance(item, str):
+                keys[item] = None
+            elif isinstance(item, dict):
+                for key, nested_items in item.items():
+                    key_base = key
+                    nested_keys = self.get_restql_query_dict(nested_items)
+                    keys[key_base] = nested_keys
+
+        return keys
+
+    def get_restql_queryset(self, queryset):
+        if self.restql_query is not None:
+            queryset = self.apply_restql_orm_mapping(queryset)
+        return queryset
```

Comment:
The `if self.restql_query is not None` bit is repeated on `get_queryset` method, I also think there is no need for this method, we can just write its contents on `get_queryset`.